### PR TITLE
Bugfix proportion aggregator

### PR DESCRIPTION
**Pull request by corinnebosley**
In numpy 1.15.2, if you construct a masked array without explicitly setting the mask to False, the array mask is a boolean value rather than an array:
```
>> import numpy.ma as ma
>> array = ma.array([1, 2, 3], mask=False)
>> array.mask
array([False, False, False])

>> array = ma.array([1, 2, 3])
>> array.mask
False
```
This is a catch for that scenario which calculates the number of unmasked points using the array itself, not the mask.

The assumption here is that if numpy returns a boolean as an array.mask, it will only ever be False.

### CODE REVIEW

**Comment by corinnebosley** [comment]
@bjlittle Tests are failing, but looks like it could be a cartopy issue causing it rather than this.

**Comment by ajdawson** [comment]
See #3230 regarding test fails.

File: lib/iris/analysis/__init__.py [hunk 1]
```diff
@@ -1237,13 +1237,20 @@ def _lazy_count(array, **kwargs):
...
     count = iris._lazy_data.non_lazy(_lazy_count)
+
     # if the incoming array is masked use that to count the total number of
     # values
     if ma.isMaskedArray(array):
         # calculate the total number of non-masked values across the given axis
-        total_non_masked = count(
-            array.mask, axis=axis, function=np.logical_not, **kwargs)
-        total_non_masked = ma.masked_equal(total_non_masked, 0)
+        if isinstance(array.mask, np.bool_):
```
[the remark this answers is not in the document]
Comment (corinnebosley) [reply lib/iris/analysis/__init__.py]: I tried `if array.mask is False`, and it doesn't catch that case.
I also tried array.mask == False, which errors when the mask is an array.

I was and still am tempted to add a case for `True` though.  Unless you come up with a better idea in the meantime I'll go with that strategy when I get back to this PR shortly.

[the remark this answers is not in the document]
Comment (ajdawson) [reply lib/iris/analysis/__init__.py]: Apologies, I meant `array.mask is np.bool_(False)` I think.

[the remark this answers is not in the document]
Comment (corinnebosley) [reply lib/iris/analysis/__init__.py]: @ajdawson thanks for the tip, that works nicely.

**Comment by ajdawson** [comment]
Looks like you have changes that don't belong here now. I guess your branch is based on master but is now targeted for merge into v2.2.x. This needs to be resolved before this can be merged.

**Comment by bjlittle** [comment]
@corinnebosley Can you cherry-pick your commit onto a branch based on `2.2.x` instead.

As @ajdawson mentions, otherwise you'll be pulling in `master` branch changes onto the `2.2.x` branch, instead of the other way around.

Thanks

**Comment by corinnebosley** [comment]
Replaced by #3233

**Pull request closed without merging**